crates/jxl-oxide/src/lib.rs

```diff
@@ -135,6 +135,7 @@ impl<R: Read> JxlImage<R> {
             image_header: &self.image_header,
             embedded_icc: self.embedded_icc.as_deref(),
             ctx,
+            render_spot_colour: true,
```

Comment:
We might need to prevent rendering spot color channels if the image is grayscale.

Reply:
I think this case is handled in `(0..grid.len()).for_each(|c| {` part of  `fn render_spot_colour`, since at this point, we have the right amount of channels in `grid`
https://github.com/tirr-c/jxl-oxide/blob/b26f211eb86e0c3cf839d63a58f0dc4216750da9/crates/jxl-oxide/src/lib.rs#L274-L275

Reply:
Сan we use `red` color from spot channel as luma and just ignore `green` and `blue` as it is now?

Reply:
I'm not sure about the semantics of spot color channels on grayscale images; bitstream does not forbid this, but tricolor spot color channel doesn't make sense with grayscale ICC profile. Let's go with emitting a warning for this time.